src/operator/tensor/matrix_op-inl.h

```diff
@@ -316,6 +316,12 @@ inline bool ExpandDimShape(const nnvm::NodeAttrs& attrs,
   ret[param.axis] = 1;
   for (index_t i = param.axis+1; i < ret.ndim(); ++i) ret[i] = shp[i-1];
   SHAPE_ASSIGN_CHECK(*out_attrs, 0, ret);
+
```

Comment:
One problem I noticed here is that param.axis of type index_t. When users enter a meaningful negative number for the axis, it would be interpreted as a very big unsigned int, which would result in CHECK_LE failure in line 312. Do you think it's a good idea to change param.axis from type index_t to int and allow negative axis input?